tools/lsp/preview.rs

```diff
@@ -69,22 +72,130 @@ struct PreviewState {
     selected: Option<element_selection::ElementSelection>,
     notify_editor_about_selection_after_update: bool,
     known_components: Vec<ComponentInformation>,
+    document_cache: Option<common::DocumentCache>,
 }
 thread_local! {static PREVIEW_STATE: std::cell::RefCell<PreviewState> = Default::default();}
 
+struct DummyWaker();
+
+impl std::task::Wake for DummyWaker {
+    fn wake(self: std::sync::Arc<Self>) { }
+}
+
+pub fn spin_once<F: std::future::Future>(future: F) -> Option<F::Output> {
```

Comment:
That's a great function. We could re-use it in some other part of Slint that currently uses spin_on.  (But not in the PR)  
Maybe the name should be "poll_once" ?